src/components/TaskList.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import "./TaskList.css";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { searchTodoSchema } from "../validation/dataSchema";
+import { SearchFormData, Task } from "../types";
+
+const TaskList = ({
+  tasks,
+  deleteTask,
+  handleCheckboxChange,
+}: {
+  tasks: Task[];
+  deleteTask: (task: Task) => void;
+  handleCheckboxChange: (id: string | undefined) => void;
+}) => {
+  const { register } = useForm<SearchFormData>({
+    resolver: zodResolver(searchTodoSchema),
+  });
+  const [searchValue, setSearchValue] = useState<string>("");
+
+  const filteredTasks = tasks.filter((task) =>
+    task.taskName.toLowerCase().includes(searchValue)
+  );
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value.toLowerCase());
```

Comment:
Here you can use a debounce to avoid trigger the function on each keyboard typing